public/js/src/module/records-queue.js

```diff
@@ -189,10 +208,12 @@ function _setUploadIntervals() {
     setTimeout(() => {
         uploadQueue();
     }, 30 * 1000);
-    // interval to check upload queued records
+    // interval to upload queued records
     setInterval(() => {
         uploadQueue();
     }, 5 * 60 * 1000);
+    // interval to disable/enable upload button
+    setInterval(_updateUploadButtonState, 0.5 * 60 * 1000);
```

Comment:
The 30 second interval was chosen as a compromise between reducing the number server requests and anxious users waiting for the button to become enabled so they can click it.

P.S. navigator.online does not work for this purpose. I think the best way to reliably check the online status is what we're currently doing with a HTTP request.